### ISSUE TITLE
Almost all glyphs disappeared from file!?

### ISSUE THREAD

**Issue by Ginkgo100** [fontforge/fontforge#2281]
This is truly bizarre. Deep breaths.

I have been working on my font for some time. It's an all-caps font, and I have all 26 uppercase letters, all 10 digits, and quite a lot of punctuation completed. The whole time, I have been saving frequently, but never made a backup of the .sfd file. I decided to generate a font file to help with kerning (the idea was to put some random text in a word processor and see how letter combinations work in different words). All went well and I successfully installed the font file.

A few nights later, my computer restarted itself for the automatic Windows 7 backup. I did not close FontForge before the backup, and my project's file was open at the time. But I was not worried because I was sure I had saved my work.

When I FontForge again, I got the message (which I expected) saying that there was a file to recover or some such. I decided to put off the recovery and just open my project's file again. 

THERE WERE ONLY TWO GLYPHS IN THE PROJECT. About 70 glyphs had gone missing. I only had D and P.

I closed FontForge and reopened it to do the file recovery (I'm sure there is a way to do this without closing it, but I was feeling a sort of... urgency at this moment). The result was that there are now four glyphs in my project (K and R were added). All the rest are still missing. I don't know why D and P were preserved, but I remember that K and R were two of the last glyphs I worked on, probably THE last two, before the restart.

I searched my computer both for the name of the file (without extension), and for *.sfd. No luck. The project file is only 4 KB.

The font file, fortunately, is still intact, so only the small changes made after I generated it have been lost (maybe not even all of those, since I think the recovered K and R glyphs in the original project file are the latest versions).

But what on earth might have happened? Can it be prevented in the future (besides backing up the project file frequently)?

**Comment by jtanx** [comment]
They might not actually have disappeared, but been moved to the background (#1918).

Assuming you're using Windows, please try using [this build](http://sourceforge.net/projects/fontforgebuilds/files/patched-builds/FontForge-mingw-w64-i686-1aba9c%2B8a9d4a-ungetc.7z/download) - it will prevent this from happening again but it won't fix it - if the glyphs have been moved to the background, you'll need to manually move it to the foreground.

**Comment by Ginkgo100** [comment]
Yes, I had been hoping at first that it was some issue where the glyphs were still there, just hidden—like their having been moved from Fore to Back, or something like that. But the missing glyphs are all reset to the em width (1000), and at 4KB the file itself is too small to have the glyphs hidden somewhere.

**Comment by larsenwork** [comment]
@Ginkgo100 can you reproduce this? If yes, then update your initial post to have just the steps needed to reproduce it - if no, then close this issue:)
